lib/ts/recipe/emailpassword/api/implementation.ts

```diff
@@ -65,7 +65,16 @@ export default function getAPIImplementation(): APIInterface {
             | { status: "PASSWORD_RESET_NOT_ALLOWED"; reason: string }
             | GeneralErrorResponse
         > {
-            const email = formFields.filter((f) => f.id === "email")[0].value;
+            // NOTE: Check for email being a non-string value. This check will likely
+            // never evaluate to `true` as there is an upper-level check for the type
+            // in validation but kept here to be safe.
+            const emailAsUnknown = formFields.filter((f) => f.id === "email")[0].value;
+            if (typeof emailAsUnknown !== "string")
+                return {
+                    status: "GENERAL_ERROR",
+                    message: "email value needs to be a string",
+                };
```

Comment:
```suggestion
                throw new Exception("Should never come here since we already check that the email value is a string in validateFormFieldsOrThrowError");
```

Reply:
similar changes everywhere else

Reply:
Okay.